### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by seanm**
These are all pretty conservative changes.

### CODE REVIEW

**Comment by seanm** [comment]
@chrisballinger @jakebromberg @jdeff @zhouzhongguang as you've all committed to this project recently, perhaps you'd care to review this?

File: README.markdown [hunk 1]
```diff
@@ -80,13 +80,13 @@ import CocoaAsyncSocket
   _It runs entirely within its own GCD dispatch_queue, and is completely thread-safe. Further, the delegate methods are all invoked asynchronously onto a dispatch_queue of your choosing. This means parallel operation of your socket code, and your delegate/processing code._
 
 - The Latest Technology & Performance Optimizations<br/>
-  _Internally the library takes advantage of technologies such as [kqueue's](http://en.wikipedia.org/wiki/Kqueue) to limit [system calls](http://en.wikipedia.org/wiki/System_call) and optimize buffer allocations. In other words, peak performance._
+  _Internally the library takes advantage of the latest technologies and optimizes buffer allocations. In other words, peak performance._
```
Comment (jakebromberg) [inline README.markdown]: "The latest technologies" has the bluff of a sales pitch. This library is in maintenance mode and doesn't incorporate any new technologies (it's based on the deprecated BSD Sockets). Why were the links removed?

File: Source/GCD/Documentation.html [hunk 2]
```diff
@@ -30,12 +30,12 @@ <h4>
 </p>
 
 <p>
-Did I mention you should read the headers? They're docemented very nicely, in plain english.
+Did I mention you should read the headers? They're docemented very nicely, in plain English.
```
Comment (jakebromberg) [inline Source/GCD/Documentation.html]: Typo: documented

File: README.markdown (hunk 1 above)
Comment (jakebromberg) [inline README.markdown]: "The latest technologies" has the bluster of a sales pitch. This project is in maintenance mode. It's based on the deprecated BSD Sockets. Why were the links removed?

File: Source/GCD/GCDAsyncUdpSocket.m [hunk 3]
```diff
@@ -474,7 +481,7 @@ - (void)dealloc
 #pragma mark Configuration
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-- (id)delegate
+- (id <GCDAsyncUdpSocketDelegate>)delegate
```
Comment (jakebromberg) [inline Source/GCD/GCDAsyncUdpSocket.m]: Remove the space after `id`

File: README.markdown (hunk 1 above)
Comment (seanm) [reply README.markdown]: "The latest technology" was already there (line above), wasn't my wording, I just reused it in the next sentence.

As per the commit message: kqueue does not seem to be used anywhere.

I could remove the whole section if people prefer.

File: Source/GCD/Documentation.html (hunk 2 above)
Comment (seanm) [reply Source/GCD/Documentation.html]: There are no doubt other typos, I don't claim to have found them all. :)

File: Source/GCD/GCDAsyncUdpSocket.m (hunk 3 above)
Comment (seanm) [reply Source/GCD/GCDAsyncUdpSocket.m]: ok.

Comment (seanm) [reply Source/GCD/GCDAsyncUdpSocket.m]: in fact, I don't agree: if you look at line 502, the setter already uses a space (wasn't me).

There are 18 instances of "id <GCD" and 38 instances of "id<GCD".

The project is not self-consistent with whitespace/formatting.

I'd be happy to make it self-consistent (in a subsequent PR) if such a patch would be accepted.

Comment (seanm) [reply Source/GCD/GCDAsyncUdpSocket.m]: Most of the 38 are probably from my changes, so if anything, I should add the space to be consistent with precedent.

File: Source/GCD/GCDAsyncSocket.m [hunk 4]
```diff
@@ -240,8 +247,7 @@ - (id)initWithCapacity:(size_t)numBytes
 
 - (void)dealloc
 {
-	if (preBuffer)
-		free(preBuffer);
+	free(preBuffer);
```
Comment (chrisballinger) [inline Source/GCD/GCDAsyncSocket.m]: I'd still prefer to have the nil check here

Comment (seanm) [reply Source/GCD/GCDAsyncSocket.m]: OK, if it will help get this merged, I'll remove it, I'm not wedded to it. But I'm curious why? The behaviour of free(null) is guaranteed by the C standard and macOS man page. Tools like clang-tidy even get rid of such checks (for C++ delete at least): https://clang.llvm.org/extra/clang-tidy/checks/readability-delete-null-pointer.html

File: README.markdown (hunk 1 above)
Comment (seanm) [reply README.markdown]: OK, I've removed the whole section.

Comment (seanm) [reply README.markdown]: OK, I've removed the whole section.

File: Source/GCD/Documentation.html (hunk 2 above)
Comment (seanm) [reply Source/GCD/Documentation.html]: Fixed that additional typo.

File: Source/GCD/GCDAsyncSocket.m (hunk 4 above)
Comment (seanm) [reply Source/GCD/GCDAsyncSocket.m]: OK, I've removed that commit, to restore the useless nil check.

File: Source/GCD/GCDAsyncUdpSocket.m (hunk 3 above)
Comment (seanm) [reply Source/GCD/GCDAsyncUdpSocket.m]: OK, removed that space. Also added another commit to make them uniform.

**Comment by seanm** [comment]
@chrisballinger @jakebromberg @jdeff @zhouzhongguang OK, I've changed everything requested so far...

Hopefully someone can review more and we can merge this...

**Comment by seanm** [comment]
@chinsyo thanks for reviewing!

**Comment by seanm** [comment]
friendly ping...

@chinsyo can this be merged now?

**Comment by chinsyo** [comment]
> friendly ping...
> 
> @chinsyo can this be merged now?

Sorry, I don't have access to merge pull request for this project.

**Comment by seanm** [comment]
Do you know who does?
